pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/TaskSchedulingContext.java

```diff
@@ -132,4 +133,9 @@ public TaskSchedulingContext setLeader(boolean leader) {
     _isLeader = leader;
     return this;
   }
+
+  public static boolean isUserTriggeredTask(String triggeredBy) {
+    return CommonConstants.TaskTriggers.MANUAL_TRIGGER.toString().equals(triggeredBy)
```

Comment:
`MANUAL_TRIGGER` sticks to the task configs provided as part of table, should we treat it in same way as `ADHOC_TRIGGER`? We don't want to end up in a place where "-1" fails in schedule but as soon as user clicks "Schedule Now" it goes through.

Reply:
"Schedule Now" will fail if it has -1 and generates more tasks than max allowed.